src/libraries/System.Formats.Nrbf/src/System/Formats/Nrbf/ArraySinglePrimitiveRecord.cs

```diff
@@ -11,37 +11,55 @@
 using System.Reflection.Metadata;
 using System.Runtime.CompilerServices;
 using System.Runtime.InteropServices;
-using System.Runtime.Serialization.BinaryFormat.Utils;
+using System.Formats.Nrbf.Utils;
 
-namespace System.Runtime.Serialization.BinaryFormat;
+namespace System.Formats.Nrbf;
 
 /// <summary>
 /// Represents a single-dimensional array of a primitive type.
 /// </summary>
 /// <remarks>
 /// ArraySinglePrimitive records are described in <see href="https://learn.microsoft.com/openspecs/windows_protocols/ms-nrbf/3a50a305-5f32-48a1-a42a-c34054db310b">[MS-NRBF] 2.4.3.3</see>.
 /// </remarks>
-internal sealed class ArraySinglePrimitiveRecord<T> : ArrayRecord<T>
+internal sealed class ArraySinglePrimitiveRecord<T> : SZArrayRecord<T>
     where T : unmanaged
 {
-    private static TypeName? s_elementTypeName;
+    private static TypeName? s_typeName;
 
     internal ArraySinglePrimitiveRecord(ArrayInfo arrayInfo, IReadOnlyList<T> values) : base(arrayInfo)
     {
         Values = values;
         ValuesToRead = 0; // there is nothing to read anymore
     }
 
-    public override RecordType RecordType => RecordType.ArraySinglePrimitive;
+    public override SerializationRecordType RecordType => SerializationRecordType.ArraySinglePrimitive;
 
-    public override TypeName ElementTypeName
-        => s_elementTypeName ??= TypeName.Parse(typeof(T).FullName.AsSpan()).WithAssemblyName(typeof(T).GetAssemblyNameIncludingTypeForwards());
+    /// <inheritdoc />
+    public override TypeName TypeName
+    {
+        get
+        {
+            TypeName? typeName = s_typeName;
+            if (typeName is null)
+            {
+                if (typeof(T) == typeof(TimeSpan))
+                {
+                    // TimeSpan is the only NRBF primitive type with no [TypeForwardedFrom("mscorlib")] annotation.
```

Comment:
It looks like an oversight. Is this special case required to make something work? I think it would make more sense to return mscorlib for this case too (ie delete this special case).

Reply:
I've removed it.